src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if ((object.getClass().equals(Rectangle.class))) {
+            Rectangle rectangle = (Rectangle) object;
+            return (rectangle.length == null && length == null
+                    || length.equals(rectangle.length))
```

Comment:
```suggestion
            return (rectangle.length == null && length == null
                    || length.equals(rectangle.length))
```
rectangle.length != null, length == null —> you will get NPE by this condition
In this case this check (rectangle.length == null) will give us false and we will gonna move to the right part length.equals(rectangle.length)) where NPE will happen;